c++/src/kj/compat/gzip-test.c++

```diff
@@ -193,6 +193,19 @@ KJ_TEST("async gzip decompression") {
         gzip.tryRead(text, 1, sizeof(text)).wait(io.waitScope));
   }
 
+  // Read corrupt input.
+  {
+    Vector<byte> bytes;
+    bytes.addAll(ArrayPtr<const byte>(FOOBAR_GZIP));
+    bytes[20] ^= 0x55;  // Corrupt gzip stream.
+    MockAsyncInputStream rawInput(bytes, kj::maxValue);
+    GzipAsyncInputStream gzip(rawInput);
+
+    char text[16]{};
+    KJ_EXPECT_THROW_MESSAGE("gzip decompression failed; ctx.msg = incorrect data check",
+        gzip.tryRead(text, 1, sizeof(text)).wait(io.waitScope));
```

Comment:
Does this test fail without the patch? I would expect the call to `tryRead()` to throw before the patch, and the `.wait()` to throw after the patch, but I'd expect the test to pass either way.